scripts/skills/perks/perk_rf_phalanx.nut

```diff
@@ -25,8 +25,16 @@ this.perk_rf_phalanx <- ::inherit("scripts/skills/skill", {
 			id = 6,
 			type = "text",
 			icon = "ui/icons/rf_reach.png",
-			text = ::MSU.Text.colorizeValue(this.getCount()) + " Reach when attacking or defending in melee"
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeValue(this.getCount()) + " [Reach|Concept.Reach] when attacking or defending in melee")
 		});
+
+		tooltip.push({
```

Comment:
I would capsule both tooltip lines in their own if to only show up while their effects are actually active.

```
if (this.getCount())
{
// Add Reach Tooltip
}

if (this.hasAdjacentShieldwall())
{
// Add Shieldwall expire tooltip
}
```

And the Shieldwall expire tooltip would then be changed into
`"[Shieldwall|Skill+shieldwall_effect] on this character will not expire"`

Reply:
I've improved the tooltip a bit based on your suggestion. I think it is important to mention the condition in the tooltip i.e. being next to an ally with shieldwall. A general statement like "shieldwall will not expire" may be misinterpreted to mean that it won't expire unconditionally.